haplotypecaller-gvcf-gatk4.wdl

```diff
@@ -144,102 +150,114 @@ task CramToBamTask {
     preemptible: select_first([preemptible_attempts, 3])
  }
   output {
-    File output_bam = "${sample_name}.bam"
-    File output_bai = "${sample_name}.bai"
+    File output_bam = "~{sample_name}.bam"
+    File output_bai = "~{sample_name}.bai"
   }
 }
 
 # HaplotypeCaller per-sample in GVCF mode
 task HaplotypeCaller {
-  File input_bam
-  File input_bam_index
-  File interval_list
-  String output_filename
-  File ref_dict
-  File ref_fasta
-  File ref_fasta_index
-  Float? contamination
-  Boolean make_gvcf
-
-  String gatk_path
-  String? java_options
-  String java_opt = select_first([java_options, "-XX:GCTimeLimit=50 -XX:GCHeapFreeLimit=10"])
-
-  # Runtime parameters
-  String docker
-  Int? mem_gb
-  Int? disk_space_gb
-  Boolean use_ssd = false
-  Int? preemptible_attempts
+  input {
+    # Command parameters
+    File input_bam
+    File input_bam_index
+    File interval_list
+    String output_filename
+    File ref_dict
+    File ref_fasta
+    File ref_fasta_index
+    Float? contamination
+    Boolean make_gvcf
+    Int hc_scatter
+
+    String gatk_path
+    String? java_options
+
+    # Runtime parameters
+    String docker
+    Int? mem_gb
+    Int? disk_space_gb
+    Boolean use_ssd = false
+    Int? preemptible_attempts
+  }
+
+  String java_opt = select_first([java_options, "-XX:GCTimeLimit=50 -XX:GCHeapFreeLimit=10"]) 
 
   Int machine_mem_gb = select_first([mem_gb, 7])
   Int command_mem_gb = machine_mem_gb - 1
 
   Float ref_size = size(ref_fasta, "GB") + size(ref_fasta_index, "GB") + size(ref_dict, "GB")
-  Int disk_size = ceil(size(input_bam, "GB") + ref_size) + 20
-
-  command <<<
-  set -e
+  Int disk_size = ceil(((size(input_bam, "GB") + 30) / hc_scatter) + ref_size) + 20
+  
+  parameter_meta {
+    input_bam: {
+      description: "a bam file",
+      localization_optional: true
+    }
+    input_bam_index: {
+      description: "an index file for the bam input",
+      localization_optional: true
+    }
+  }
+  command {
+    set -e
   
-    ${gatk_path} --java-options "-Xmx${command_mem_gb}G ${java_opt}" \
+    ~{gatk_path} --java-options "-Xmx~{command_mem_gb}G ~{java_opt}" \
       HaplotypeCaller \
-      -R ${ref_fasta} \
-      -I ${input_bam} \
-      -L ${interval_list} \
-      -O ${output_filename} \
-      -contamination ${default=0 contamination} ${true="-ERC GVCF" false="" make_gvcf}
-  >>>
-
+      -R ~{ref_fasta} \
+      -I ~{input_bam} \
+      -L ~{interval_list} \
+      -O ~{output_filename} \
+      -contamination ~{default=0 contamination} ~{true="-ERC GVCF" false="" make_gvcf}
```

Comment:
Admittedly our production GVCF WDLs are different for exomes and genomes, but can we add allele-specific annotations here?

Reply:
Does this mean simply appending the following line to the command ?
`-G StandardAnnotation -G AS_StandardAnnotation -G StandardHCAnnotation`
[source](https://gatk.broadinstitute.org/hc/en-us/articles/360035890551?id=9622)

Reply:
Exactly.